From 1bbe4f4c985819c8f8ead54288ca1cc8a4b7f51f Mon Sep 17 00:00:00 2001
Message-Id: <1bbe4f4c985819c8f8ead54288ca1cc8a4b7f51f.1476965418.git.keller@fs-net.de>
From: Hartmut Keller <keller@fs-net.de>
Date: Wed, 19 Oct 2016 15:28:28 +0200
Subject: [PATCH 01/12] os-compatibility: define CLOCK_BOOTTIME when not
 available

CLOCK_BOOTTIME is supposed to be provided by the C library headers,
but uClibc 0.9.33.2 does not provide it. Instead of depending on an
uClibc patch, let's make Weston define it to the right value when not
already defined by the C library.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 shared/os-compatibility.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/shared/os-compatibility.h b/shared/os-compatibility.h
index 06d2574..d6b7667 100644
--- a/shared/os-compatibility.h
+++ b/shared/os-compatibility.h
@@ -29,6 +29,7 @@
 #include "config.h"
 
 #include <sys/types.h>
+#include <time.h>
 
 #ifdef HAVE_EXECINFO_H
 #include <execinfo.h>
@@ -40,6 +41,10 @@ backtrace(void **buffer, int size)
 }
 #endif
 
+#ifndef CLOCK_BOOTTIME
+#define CLOCK_BOOTTIME 7
+#endif
+
 int
 os_fd_set_cloexec(int fd);
 
-- 
1.7.4.4

